esp/services/ws_dfu/ws_dfuService.cpp

```diff
@@ -5885,4 +5897,207 @@ int CWsDfuEx::GetIndexData(IEspContext &context, bool bSchemaOnly, const char* i
+{
+    metaInfo.setProp("version", securityInfoVersion);
+    metaInfo.setProp("logicalFilename", file.queryLogicalName());
+    metaInfo.setProp("jobId", req.getJobId());
+    metaInfo.setProp("accessType", req.getAccessTypeAsString());
+    StringBuffer userStr;
+    if (user)
+        metaInfo.setProp("user", user->getUserName(userStr).str());
+
+    const char *clusterName = req.getCluster(); // can be null
+    Owned<IFileDescriptor> fDesc = file.getFileDescriptor(clusterName);
+
+#ifdef _USE_OPENSSL
+    if (keyPairName) // without it, meta data is not encrypted
+    {
+        metaInfo.setProp("keyPairName", keyPairName);
+
+        unsigned expirySecs = req.getExpirySeconds();
+
+        // JCSMORE - make this configurable
+        if (expirySecs > 86400)
+            expirySecs = 86400; //24 hours
+
+        // setup "expiryTime"
+        time_t simple;
+        time(&simple);
+        simple += expirySecs;
+        CDateTime expiryDt;
+        expiryDt.set(simple);
+        StringBuffer expiryTimeStr;
+        expiryDt.getString(expiryTimeStr);
+        Owned<IPropertyTree> secureMetaInfo = createPTreeFromIPT(&metaInfo);
+        secureMetaInfo->setProp("expiryTime", expiryTimeStr);
+
+        extractFilePartInfo(*secureMetaInfo, *fDesc);
+
+        // create random AES key and IV
+        char randomAesKey[aesKeySize];
+        char randomIV[aesBlockSize];
+        fillRandomData(aesKeySize, randomAesKey);
+        fillRandomData(aesBlockSize, randomIV);
+
+        // 1st serialize to JSON
+        StringBuffer secureMetaInfoJson;
+        toJSON(secureMetaInfo, secureMetaInfoJson);
+
+        // 2nd compress
+        MemoryBuffer compressedSecureMetaInfoMb;
+        fastLZCompressToBuffer(compressedSecureMetaInfoMb, secureMetaInfoJson.length(), secureMetaInfoJson.str());
+
+        // 3rd encrypt with AES key
+        MemoryBuffer encryptedSecureMetaInfoMb;
+        aesKeyEncrypt(encryptedSecureMetaInfoMb, compressedSecureMetaInfoMb.length(), compressedSecureMetaInfoMb.bytes(), randomAesKey, randomIV);
+
+        metaInfo.setPropBin("secureInfo", encryptedSecureMetaInfoMb.length(), encryptedSecureMetaInfoMb.bytes());
+
+        // 4th encrypt AES key with public *DAFILESRV* key
+        VStringBuffer publicKeyFName("%s.pub.pem", keyPairName); // JCSMORE - could cache loaded keys
+        Owned<CLoadedKey> publicKey = loadPublicKeyFromFile(publicKeyFName, nullptr);
+        MemoryBuffer encryptedAesKeyMb;
+        publicKeyEncrypt(encryptedAesKeyMb, aesKeySize, randomAesKey, *publicKey);
+
+        /* perhaps should combine into single prob..
+         * Does IV need to be generated/sent each time?
+         */
+        metaInfo.setPropBin("securityKey", encryptedAesKeyMb.length(), encryptedAesKeyMb.bytes());
+        metaInfo.setPropBin("securityIV", aesBlockSize, randomIV);
+    }
+    else
```

Comment:
The idea is that it will only take this path if : 
a) A system without OPENSSL is built, OR
b) The system (or part of it) has been configured without FileAccess security, in which case the getFileDafilesrvKeyName() will return no [configured] key pair name, and security is disabled for that cluster/environment.